pkg/client/client.go

```diff
@@ -0,0 +1,9 @@
+package client
+
+import k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+//go:generate mockgen -source=client.go -package=client -destination=mock_client.go
+
+type Client interface {
```

Comment:
Not sure this type is needed - `gomock` can directly generate a mock for `sigs.k8s.io/controller-runtime/pkg/client.Client`.

Reply:
i think that it is a good practice to create a Client layer for accessing the cluster: that way we can use whatever client we need (besides controller-runtime), without changing the rest of the code. As we saw in SRO , we needed to use more then one client implementation

Reply:
Actually, I don't see any benefit in using several client implementations - controller-runtime's is generic enough and works with all object types.

Reply:
> gomock can directly generate a mock for sigs.k8s.io/controller-runtime/pkg/client.Client

@qbarrand How do you do that without adding the mockgen annotation to `sigs.k8s.io/controller-runtime/pkg/client.Client`?

Reply:
@ybettan in any file:
```
//go:generate mockgen -destination mock_client.go sigs.k8s.io/controller-runtime/pkg/client Client
```

Reply:
The idea of this internal type was because I didn't know how to specify only one interface and the `mockgen` was generating mocks for 7 different interfaces. Now, as @qbarrand indicated, we can specify only the needed interface `Client`, so I would go with that approach instead of creating an internal type :+1: